opal/mca/base/mca_base_framework.c

```diff
@@ -255,19 +256,18 @@ int mca_base_framework_close(struct mca_base_framework_t *framework)
         if (OPAL_SUCCESS != ret) {
             return ret;
         }
-    } else {
-        opal_list_item_t *item;
-        while (NULL != (item = opal_list_remove_first(&framework->framework_components))) {
-            mca_base_component_list_item_t *cli;
-            cli = (mca_base_component_list_item_t *) item;
-            mca_base_component_unload(cli->cli_component, framework->framework_output);
-            OBJ_RELEASE(item);
-        }
-        while (NULL != (item = opal_list_remove_first(&framework->framework_failed_components))) {
-            OBJ_RELEASE(item);
-        }
-        ret = OPAL_SUCCESS;
     }
+    opal_list_item_t *item;
```

Comment:
I am not sure I agree with this change, as it really changes the behavior of the unloading. The component unloading is now done explicitly for all components, instead of delegating that task to the framework itself.

Reply:
Ok.  If this is not the solution, then at least the timer framework close function is incorrect -- i.e., it will need to do the component unloading and emptying of the framework component list.  Currently: the timer framework close function basically does one special thing (specific to timers) and then returns.

I did not check to see if other frameworks were in the same situation (i.e., have a framework close function).

Reply:
We already unload the components in `mca_base_component_close` and `mca_base_components_close`.

Looking at the true branch of the `if` it seems that the only case not covered is when the framework has its own `close` and that `close` does not clean behind. Thus, either we fix there or we check this patch thoroughly to make sure it does not break anything else and does not lead to double unloads of components.

Reply:
Oops!  I missed the `if (is_open)` surrounding condition -- my eye zoomed on the "if there's a framework_close" conditional.  Now I see what you're saying.  Let me re-work this patch.

Reply:
@bosilca Re-worked.  Have a look now.